### PR TITLE
Migrate most tests away from runtime check of build flavor

### PR DESCRIPTION
**Pull request by prprabhu-ms**
This PR migrates most browser tests off of runtime checks for build flavor (via `isTestProfileStableFlavor`) to instead use conditional compilation.

The benefit is that the tests will not have to be separately modified when the relevant feature is stabilized.

### CODE REVIEW

File: common/config/env/.env [hunk 1]
```diff
@@ -1,2 +1,2 @@
 // THIS IS AN AUTO GENERATED FILE BY `RUSH SWITCH <BETA|STABLE>`. DO NOT MODIFY
-COMMUNICATION_REACT_FLAVOR=beta
```
Comment (prprabhu-ms) [inline common/config/env/.env:2]: Err.... NO.

File: packages/react-composites/tests/browser/call/hermetic/HoldScreen.test.ts [hunk 2]
```diff
@@ -3,13 +3,12 @@
 
 import { expect } from '@playwright/test';
 import { IDS } from '../../common/constants';
-import { dataUiId, isTestProfileStableFlavor, pageClick, stableScreenshot, waitForSelector } from '../../common/utils';
+import { dataUiId, pageClick, stableScreenshot, waitForSelector } from '../../common/utils';
 import { buildUrlWithMockAdapter, defaultMockCallAdapterState, defaultMockRemoteParticipant, test } from './fixture';
 
+/* @conditional-compile-remove(PSTN-calls) @conditional-compile-remove(one-to-n-calling) */
```
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/call/hermetic/HoldScreen.test.ts:9]: I thought when denoting 2 features in a CC statement it was something like `@conditional-compile-remove(feature 1 feature 2)` when in the same comment?

File: packages/react-composites/tests/browser/callwithchat/live/CallWithChatComposite.test.ts [hunk 3]
```diff
@@ -54,17 +53,17 @@ test.describe('CallWithChat Composite CallWithChat Page Tests', () => {
...
-    test.skip(isTestProfileStableFlavor());
     const page = pages[1];
     await pageClick(page, dataUiId('call-with-chat-composite-more-button'));
     const moreButtonShowDialpadButton = await page.$('div[role="menu"] >> text="Show dialpad"');
     await moreButtonShowDialpadButton?.click();
     expect(await stableScreenshot(page)).toMatchSnapshot(`call-with-chat-more-drawer-dtmf-dialpad.png`);
   });
 
+  /* @conditional-compile-remove(PSTN-calls) */
   test('More Drawer menu opens and can choose to be on hold', async ({ pages }) => {
```
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/callwithchat/live/CallWithChatComposite.test.ts:66]: this test is also for 1-n-calling

**Review by dmceachernmsft — APPROVED** [review]

File: common/config/env/.env (hunk 1 above)
Comment (JamesBurnside) [reply common/config/env/.env]: out of interest, this should be gitignored:
image

**Review by anjulgarg — APPROVED** [review]

File: packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts [hunk 4]
```diff
@@ -7,16 +7,15 @@ import {
...
   isTestProfileDesktop,
-  isTestProfileStableFlavor,
   pageClick,
   stableScreenshot,
   waitForSelector
 } from '../../common/utils';
 import { buildUrlWithMockAdapter, defaultMockCallAdapterState, defaultMockRemoteParticipant, test } from './fixture';
 
 test.describe('Participant pane tests', async () => {
+  /* @conditional-compile-remove(one-to-n-calling) */
```
Comment (edwardlee-msft) [inline packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts:17]: People Pane should actually be conditionally compiled for both PSTN and one-to-n-calling.

Comment (edwardlee-msft) [reply packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts:17]: I will go back and add the missing conditional compilation for PeoplePane related code. But in the meantime please also add PSTN here :)

Comment (edwardlee-msft) [reply packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts:17]: https://github.com/Azure/communication-ui-library/pull/2493 for tracking

File: packages/react-composites/tests/browser/chat/hermetic/FileSharing.test.ts [hunk 5]
```diff
@@ -2,12 +2,11 @@
 // Licensed under the MIT license.
 import { expect } from '@playwright/test';
 import { sendMessage, waitForMessageDelivered } from '../../common/chatTestHelpers';
-import { dataUiId, isTestProfileStableFlavor, stableScreenshot, waitForSelector } from '../../common/utils';
+import { dataUiId, stableScreenshot, waitForSelector } from '../../common/utils';
 import { buildUrlForChatAppUsingFakeAdapter, DEFAULT_FAKE_CHAT_ADAPTER_ARGS, test, TEST_PARTICIPANTS } from './fixture';
 
+/* @conditional-compile-remove(file-sharing) */
```
Comment (PorterNan) [inline packages/react-composites/tests/browser/chat/hermetic/FileSharing.test.ts:8]: How does this work? Do we add these test.ts into conditional compilation?

Comment (PorterNan) [reply packages/react-composites/tests/browser/chat/hermetic/FileSharing.test.ts:8]: Just checked the config file - we do make unittest works for conditional compilation,  but not for e2e test

**Review by PorterNan — APPROVED** [review]

**Review by edwardlee-msft — APPROVED** [review]

File: packages/react-composites/tests/browser/call/hermetic/HoldScreen.test.ts (hunk 2 above)
Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/call/hermetic/HoldScreen.test.ts:9]: Nah, we don't support fancy expressions inside the directive. But we do support multiple directives in the same / different comment.

See https://github.com/Azure/communication-ui-library/blob/b7666daf66fdb29053a89ee516a839ece5a7d9b1/packages/acs-ui-common/src/conditional-compilation-sample/index.tsx#L233

**Pull request merged**